src/dev/getting_started.md

```diff
@@ -83,6 +83,65 @@ required) for this directory instead of the default Rust compiler.
 Then you can build and test the `yk` repo using `cargo xtask` commands, for
 example `cargo xtask test`.
 
+## Rust Analyzer
+
+[Rust-analyzer](https://github.com/rust-analyzer/rust-analyzer) (RA hencforth)
+is a language server implementation for Rust.
+
+Due to [the workspace model of the `yk` repo](../tech/yk_structure.md) and the
+`x.py` build system in `ykrustc`, RA won't work out of the box. Configuration
+is dependent on the LSP plugin used, but in general:
+
+ - For the `ykrustc` repo, you need to tell RA to run `x.py` and not `cargo`.
+   This is done via the `rust-analyzer.checkOnSave.overrideCommand` option.
+
+ - For the `yk` repo, you need to point RA at the extra internal workspace via
+the `rust-analyzer.linkedProjects` option.
+
+For example, for [vim-lsp](https://github.com/prabirshrestha/vim-lsp) you would
+use a configuration similar to the following:
+
+```
+if executable('rust-analyzer')
+    if filereadable("x.py")
+        " Rust compiler.
+        au User lsp_setup call lsp#register_server({
+                    \ 'name': 'rust-analyzer',
+                    \ 'cmd': {server_info->['rust-analyzer']},
+                    \ 'allowlist': ['rust'],
+                    \ 'workspace_config': {'rust-analyzer': {'checkOnSave': {'overrideCommand': './x.py check --json-output'}}},
+                    \ })
+    elseif filereadable("ykshim_client/Cargo.toml")
+        " It's the yk repo.
+        au User lsp_setup call lsp#register_server({
+                    \ 'name': 'rust-analyzer',
+                    \ 'cmd': {server_info->['rust-analyzer']},
+                    \ 'allowlist': ['rust'],
+                    \ 'workspace_config': {'rust-analyzer': {'linkedProjects': ['internal_ws/Cargo.toml']}},
+                    \ })
+    else
+        " Normal project.
+        "
+        " Either install rust-src from rustup or set RUST_SRC_PATH to the
+        " `library` subdir of a rust source clone. If you are working on
+        " Yorick, then that's the `library` directory in your ykrustc clone.
+        au User lsp_setup call lsp#register_server({
+                    \ 'name': 'rust-analyzer',
+                    \ 'cmd': {server_info->['rust-analyzer']},
+                    \ 'allowlist': ['rust'],
+                    \ })
+    endif
+endif
+```
+
+Once rust-analyzer is enabled, you should make sure that your editor uses the
+same `RUSTFLAGS` as building and testing do. Failure to do so will cause the
+incremental build cache to be repeatedly invalidated and you'll constantly be
+rebuilding the same packages.
+
+For vim-based LSP plugins, this is just a matter of exporting `RUSTFLAGS`
```

Comment:
I'm not sure this paragraph adds anything (and indeed it's confusing, because it seems to suggest exporting an existing env var?).